pkg/promclient/promclient_e2e_test.go

```diff
@@ -142,3 +143,18 @@ func TestRule_UnmarshalScalarResponse(t *testing.T) {
 	vectorResult, err = convertScalarJSONToVector(invalidDataScalarJSONResult)
 	testutil.NotOk(t, err)
 }
+
+func TestParseVersion(t *testing.T) {
```

Comment:
All good now but I was talking about checking the parsing of different strings which might be returned by different Prometheus versions. Ideally, we would run the parsing test on all of the different, supported Prometheus versions with and without the full `/version` end-point. However, that would take a long time in the CI process, and you'd have to expand the `Makefile`, so you could probably make an array here with different, possible strings returned by `/version`. Then, we could make a logical conclusion that this functionality should be fine with the respective versions of Prometheus. Care to update this test? 😄